ui/src/main/java/org/jtaccuino/core/ui/api/CellData.java

```diff
@@ -155,8 +137,37 @@ public static OutputType of(String outputType) {
             }
         }
 
-        public static OutputData of(OutputType outputType, Map<String, String> mimeBundle) {
-            return new OutputData(outputType, mimeBundle);
+        public OutputType type();
+    }
+
+    public static record MimeTypeBasedOutputData(OutputType type, Map<String, String> mimeBundle) implements OutputData {
+
+        public static enum MimeType {
+            TEXT_PLAIN("text/plain"),
+            IMAGE_PNG("image/png");
+
+            private final String mimeType;
+
+            public String toMimeType() {
+                return mimeType;
+            }
+
+            MimeType(String mimeType) {
+                this.mimeType = mimeType;
+            }
```

Comment:
nitpick: constructor should come before methods.
```suggestion
            MimeType(String mimeType) {
                this.mimeType = mimeType;
            }

            public String toMimeType() {
                return mimeType;
            }
```

Reply:
Correct. No further change introduced here. Trying to keep the change small. Something for another PR.

Reply:
This doesn't concern pre-existing code unchanged by this PR but code introduced by this PR

Reply:
This was just moved around.